docs/explanation/dss-arch.rst

```diff
@@ -0,0 +1,152 @@
+
+MLflow
+^^^^^^
+
+`MLflow <https://ubuntu.com/blog/what-is-mlflow>`_ operates in `local mode <https://mlflow.org/docs/latest/tracking.html#other-configuration-with-mlflow-tracking-server>`_, 
+meaning that metadata and artefacts are, by default, stored in a local directory.
+
+This local directory is backed by a persistent volume, mounted to a Hostpath directory of the MLflow Pod.
+The persistent volume can be found in the directory `/mlruns`.
+
+.. _orch_layer:
+
+Orchestration
+~~~~~~~~~~~~~
+
+DSS requires a container orchestration solution. 
+DSS relies on `MicroK8s`_, a lightweight Kubernetes distribution.
+
+Therefore, MicroK8s needs to be deployed before installing DSS on the host machine. 
+It must be configured with the storage add-on. 
+This is required to use Hostpath storage in the cluster. 
+See :ref:`set_microk8s` to learn how to install MicroK8s.
+
+DSS can run with or without the use of GPUs.
+If :ref:`gpu_support` is needed, MicroK8s can be configured with the desired `GPU add-on <https://microk8s.io/docs/addon-gpu>`_. 
+
+.. _gpu_support:
+
+GPU support
+^^^^^^^^^^^
+DSS is designed to support the deployment of containerised GPU workloads on NVIDIA GPUs. 
+MicroK8s simplifies the GPU access and usage through the `NVIDIA GPU Operator <NVIDIA Operator_>`_. 
+
+DSS does not automatically install the tools and libraries required for running GPU workloads.
+To do so, it relies on MicroK8s for the required operating-system drivers.
+It also relies on the chosen image, for example, CUDA when working with NVIDIA GPUs.
+
+.. caution::
+   GPUs from other silicon vendors rather than NVIDIA can be configured. However, its functionality is not guaranteed.
+ 
+Storage
+^^^^^^^
+
+DSS expects a default `storage class <https://kubernetes.io/docs/concepts/storage/storage-classes/>`_ in the Kubernetes deployment, which is used to persist Jupyter Notebooks and MLflow artefacts.   
+In MicroK8s, the Hostpath storage add-on is chosen, used to provision Kubernetes' *PersistentVolumeClaims* (`PVCs <https://kubernetes.io/docs/concepts/storage/persistent-volumes/>`_). 
+
+A shared PVC is used across all Jupyter Notebooks to share and persist data. 
+MLflow also uses its dedicated PVC to store the logged artefacts.
+This is the DSS default storage configuration and cannot be altered.
+
+This choice ensures that all storage is backed up on the host machine in the event of MicroK8s restarts.
+
+.. note::
+   By default, you can access this storage anytime under your local directory `/var/snap/microk8s/common/default-storage`. 
+
+The following diagram summarises the DSS storage:
+
+.. figure:: https://assets.ubuntu.com/v1/2130fd66-dss_storage.png
+   :width: 800px
+   :alt: text
+
+   Storage overview
+
+.. _os_layer:
+
+Operating system
+~~~~~~~~~~~~~~~~
+
+DSS is native on Ubuntu, being developed, tested and validated on it. 
+Moreover, the solution can be used on any Linux distribution.
+
+Namespace configuration
```

Comment:
Let me know about this section, if you think it is not actually needed we can get rid of it

Reply:
The only added value of this section is that we say that one namespace is for dss and the other one is for gpu operator.

Reply:
I keep that info then, rest goes away.